lib/logstash/outputs/elasticsearch/http_client/pool.rb

```diff
@@ -66,12 +67,19 @@ def initialize(logger, adapter, initial_urls=[], options={})
       # Holds metadata about all URLs
       @url_info = {}
       @stopping = false
-    end
 
-    def oss?
-     LogStash::Outputs::ElasticSearch.oss?
+      if options[:license_check_class].nil?
+        # if a license_check_class option was not provided use the plugin local
+        # license checker class in logstash/outputs/elasticsearch/license_check.rb
+        # normally this class is passed from the Commons#build_client method to allow
+        # alternate license checking logic form other implementations like the data streams output.
+        require "logstash/outputs/elasticsearch/license_check"
+        @license_checker = LogStash::ElasticSearchOutputLicenseChecker.new(self, logger)
```

Comment:
This may break Logstash core's central pipeline management, which already reaches into the Elasticsearch Output to `build_client` without any args. I think we need to have a no-op _always-success_ implementation if no checker is given.